### DISCUSSION TITLE
Operation not supported on socket

### DISCUSSION THREAD

**Discussion by nwhobart** [Homebrew/discussions#2472] [Casks] [answered]
While trying to do a simple uninstall of the `sdm` cask I get: 

```
==> Uninstalling artifact of class Cask::Artifact::App
Warning: It seems there is already an App at '/usr/local/Caskroom/sdm/15.27.0/SDM.app'; overwriting.
==> Removing App '/usr/local/Caskroom/sdm/15.27.0/SDM.app'
==> Backing App 'SDM.app' up to '/usr/local/Caskroom/sdm/15.27.0/SDM.app'
/bin/cp -pR /Applications/SDM.app /usr/local/Caskroom/sdm/15.27.0/SDM.app
cp: /Applications/SDM.app/Contents/Resources/sdm-socket: Operation not supported on socket
Error: Failure while executing; `/bin/cp -pR /Applications/SDM.app /usr/local/Caskroom/sdm/15.27.0/SDM.app` exited with 1. Here's the output:
cp: /Applications/SDM.app/Contents/Resources/sdm-socket: Operation not supported on socket
```

I'm very confused by the error message. I'm curious if anyone has any insights or can reproduce with a simple `brew uninstall sdm`

**Comment by neutric** [comment]
I can reproduce and opened a [PR](https://github.com/Homebrew/homebrew-cask/pull/114508) to fix it. Sockets are special files for inter-process communication that cannot be copied. `sdm` does not close the socket on regular quit which leads to the error above when `uninstall` does the backup step. `rm /Applications/SDM.app/Contents/Resources/sdm-socket` and your uninstall should work smoothly.

**Comment by neutric** [comment, marked as answer]
@nwhobart Fix is merged. Please mark this as answered.
